diff --git a/pulsar/core/sally.py b/pulsar/core/sally.py
index de8a798..9b8d984 100644
--- a/pulsar/core/sally.py
+++ b/pulsar/core/sally.py
@@ -60,7 +60,7 @@ def rawWriteText(messages, path):
     rawFile = open(rawFileName, "w")
     rawQuotedFile = open("%s.rawquoted" % path, "w")
     for m in messages:
-        raw = urllib.parse.unquote(m.msg)
+        raw = urllib.parse.unquote(m.msg.decode('utf8'))
         # we split the message by standard whitespace characters
         # an quote %00 bytes...
         tokens = [presallyQuote(t) for t in raw.split()]
diff --git a/pulsar/core/util.py b/pulsar/core/util.py
index 96ef63c..e12006e 100644
--- a/pulsar/core/util.py
+++ b/pulsar/core/util.py
@@ -49,8 +49,8 @@ def scanTokens(msg, whitespace):
 
 def readDerrick(path):
         g = GzipFile(path, "rb")
-        messages = [] 
+        messages = []
         for l in g:
-            messages.append(urllib.parse.unquote(l.rstrip(b"\r\n").split(b" ", 4)[-1]))
+            messages.append(urllib.parse.unquote(l.rstrip(b"\r\n").split(b" ", 4)[-1].decode('utf8')))
         g.close()
         return messages
